### PR TITLE
build-erlang-mk: add support for buildFlags, preserve buildInputs

### PR DESCRIPTION
**Pull request by berbiche**
###### Motivation for this change

`buildErlangMk` should use `buildFlags`/`makeFlags` as all other `make`-based functions support.
The function also failed to use the user's `buildInputs`.

The first commit fixes `buildFlags` usage, I'm not sure whether `configureFlags` is also relevant.
The second commit fixes not using the passed-in `buildInputs`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by berbiche** [comment]
The second commit makes `buildErlangMk` preserve the passed-in `buildInputs`.

**Review by vikanezrimaya — COMMENTED** [review]
cc @ericbmerritt as the creator of this file - I don't think I'm able to review this without knowing what Erlang even is :laughing:

File: pkgs/development/beam-modules/build-erlang-mk.nix [hunk 1]
```diff
@@ -39,9 +40,13 @@ let
     ''
     else setupHook;
 
-    buildInputs = [ erlang perl which gitMinimal wget ];
+    buildInputs = buildInputs ++ [ erlang perl which gitMinimal wget ];
     propagatedBuildInputs = beamDeps;
 
+    buildFlags = [ "SKIP_DEPS=1" ]
+      ++ lib.optional (enableDebugInfo || erlang.debugInfo) ''ERL_OPTS="$ERL_OPTS +debug_info"''
```
Comment (SuperSandro2000) [inline pkgs/development/beam-modules/build-erlang-mk.nix:47]: I am not to much into erlang but for a release build it sounds suspicious to have debug info on.

Comment (berbiche) [reply pkgs/development/beam-modules/build-erlang-mk.nix:47]: `enableDebugInfo` defaults to false and `erlang.debugInfo` probably does too.

**Review by glittershark — APPROVED** [review]
Seems fine to me, given the relatively little I know about non-mix erlang stuff

**Review by timokau — APPROVED** [review]
Thanks!

**Pull request merged**